src/playlist-maker.js

```diff
@@ -133,217 +122,336 @@ const factory = (player, initialList, initialIndex = 0) => {
+        list[playlist.currentIndex_]
+      );
+    } else {
+      playlist.currentIndex_ = playlist.indexOf(playlist.player_.currentSrc() || '');
+    }
 
-    /**
-     * Get or set the current item in the playlist.
-     *
-     * @param  {number} [index]
-     *         If given as a valid value, plays the playlist item at that index.
-     *
-     * @return {number}
-     *         The current item index.
-     */
-    currentItem(index) {
-      if (
-        typeof index === 'number' &&
-        playlist.currentIndex_ !== index &&
-        index >= 0 &&
-        index < list.length
-      ) {
-        playlist.currentIndex_ = index;
-        playItem(
-          playlist.player_,
-          playlist.autoadvance_.delay,
-          list[playlist.currentIndex_]
-        );
-      } else {
-        playlist.currentIndex_ = playlist.indexOf(playlist.player_.currentSrc() || '');
-      }
+    return playlist.currentIndex_;
+  };
 
-      return playlist.currentIndex_;
-    },
+  /**
+   * Checks if the playlist contains a value.
+   *
+   * @param  {string|Object|Array} value
+   *         The value to check
+   *
+   * @return {boolean}
+   *         The result
+   */
+  playlist.contains = (value) => {
+    return playlist.indexOf(value) !== -1;
+  };
 
-    /**
-     * Checks if the playlist contains a value.
-     *
-     * @param  {string|Object|Array} value
-     *         The value to check
-     *
-     * @return {boolean}
-     *         The result
-     */
-    contains(value) {
-      return playlist.indexOf(value) !== -1;
-    },
+  /**
+   * Gets the index of a value in the playlist or -1 if not found.
+   *
+   * @param  {string|Object|Array} value
+   *         The value to find the index of
+   *
+   * @return {number}
+   *         The index or -1
+   */
+  playlist.indexOf = (value) => {
+    if (typeof value === 'string') {
+      return indexInSources(list, value);
+    }
 
-    /**
-     * Gets the index of a value in the playlist or -1 if not found.
-     *
-     * @param  {string|Object|Array} value
-     *         The value to find the index of
-     *
-     * @return {number}
-     *         The index or -1
-     */
-    indexOf(value) {
-      if (typeof value === 'string') {
-        return indexInSources(list, value);
+    const sources = Array.isArray(value) ? value : value.sources;
+
+    for (let i = 0; i < sources.length; i++) {
+      const source = sources[i];
+
+      if (typeof source === 'string') {
+        return indexInSources(list, source);
+      } else if (source.src) {
+        return indexInSources(list, source.src);
       }
+    }
 
-      const sources = Array.isArray(value) ? value : value.sources;
+    return -1;
+  };
 
-      for (let i = 0; i < sources.length; i++) {
-        const source = sources[i];
+  /**
+   * Get the index of the last item in the playlist.
+   *
+   * @return {number}
+   *         The index of the last item in the playlist.
+   */
+  playlist.indexOfLast = () => {
```

Comment:
maybe it would be better to call this `lastIndex`?

Reply:
Should their be a `currentIndex` as well? seems to be missing. EDIT I see that currentItem returns an index rather than an item, I guess that could work, but it does seem a bit weird.

Reply:
It's because `currentItem()` will look at the player source to determine if it can update the `currentIndex_`; so, it's the best way to get the current index.

Reply:
As for naming... I dunno. Seems a little bike-sheddy, but the choice here was to stay consistent with the naming scheme provided by `indexOf`.